proto/xmtpv4/envelopes/envelopes.proto

```diff
@@ -60,3 +60,45 @@ message OriginatorEnvelope {
     BlockchainProof blockchain_proof = 3;
   }
 }
+
+enum Misbehavior {
+  MISBEHAVIOR_UNSPECIFIED = 0;
+  MISBEHAVIOR_UNRESPONSIVE_NODE = 1;
+  MISBEHAVIOR_SLOW_NODE = 2;
+  MISBEHAVIOR_FAILED_REQUEST = 3;
+  MISBEHAVIOR_OUT_OF_ORDER = 4;
+  MISBEHAVIOR_DUPLICATE_SEQUENCE_ID = 5;
+  MISBEHAVIOR_CAUSAL_ORDERING = 6;
+  MISBEHAVIOR_INVALID_PAYLOAD = 7;
+  MISBEHAVIOR_BLOCKCHAIN_INCONSISTENCY = 8;
+}
+
+message LivenessFailure {
+  uint32 response_time_ns = 1;
+  SubscribeEnvelopesRequest subscribe = 2;
+  QueryEnvelopesRequest query = 3;
+  PublishPayerEnvelopesRequest publish = 4;
+}
+
+message SafetyFailure {
+  repeated OriginatorEnvelope envelopes = 1;
+}
+
+message UnsignedMisbehaviorReport {
+  uint64 reporter_time_ns = 1;
+  uint32 misbehaving_node_id = 2;
+  Misbehavior type = 3;
+  LivenessFailure liveness = 4;
+  SafetyFailure safety = 5;
+  // Nodes must verify this field is false for client-submitted reports
+  bool submitted_by_node = 6;
+}
+
+message MisbehaviorReport {
+  // Server time when the report was stored. Used only for querying reports.
+  // This field is not signed.
+  uint64 server_time_ns = 1;
+  bytes unsigned_misbehavior_report = 2;
+  // Signed by the node hosting the report
+  xmtp.identity.associations.RecoverableEcdsaSignature signature = 3;
```

Comment:
We have the node's signature here. Alternative approaches:

- No signature - it makes it a bit tricky to handle cases where a node retracts or denies a report it previously made. It also seems useful to know which node client reports came in through, as opposed to them going into one big pool.
- Add a client signature - I think in most cases clients will want to stay anonymous. Adding your public key to a set of group payloads may end up deanonymizing a group.

When we define incentives for submitting reports, some entities may want to be rewarded. When we get there, perhaps we could add a `claimant_public_key` field to the `UnsignedMisbehaviorReport`. Probably don't need an additional signature for that.